templates/factory/pages/demo/endpoint.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState } from "react";
+
+import { DoubleArrowRightIcon } from "@radix-ui/react-icons";
+
+import { useCodeflyContext } from "../providers/codefly.provider";
+import { useResponseData } from "./response.provider";
+
+
+const Endpoint = ({ endpoint }) => {
+    const { routing } = useCodeflyContext();
+    const { setResponse, setEndpoint, setLoading, setRoute } = useResponseData();
+
+    const hanldeFetch = async (route, data) => {
+        const { method, path } = route;
+
+        const url = routing(method, endpoint.service, path)
+
+        try {
+            if (!url) {
+                return;
+            }
+
+            setLoading(true);
+            const response = await fetch("http://" + url, {
+                method: method,
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: method !== "GET" ? JSON.stringify(data) : null,
+            });
+            const responseData = await response.json();
+            setResponse(responseData);
+            setLoading(false);
+
+        } catch (error) {
+            console.error('Error posting data:', error);
+        }
+    };
+
+    const [isContentVisible, setIsContentVisible] = useState<boolean>(false);
+
+    const handleRouteClick = (route) => {
+        if (route.method === 'GET') {
+
+            hanldeFetch(route, null)
+            setRoute(route);
+        }
+    }
+
+    const handleEndpointClick = () => {
+        setIsContentVisible(!isContentVisible);
+        setEndpoint(endpoint)
+    }
+
+    return (
+        <div className="bg-gray-200 dark:bg-gray-800 p-0 mb-4 cursor-pointer hover:bg-gray-300 border m-8 mt-0 border-gray-400 rounded-md">
```

Comment:
kept with tailwindcss as it's used by nextjs it self.

not a big fan of this long strings